src/telem/lib/input/mekanism/BioGeneratorInputAdapter.lua

```diff
@@ -0,0 +1,70 @@
+local o = require 'telem.lib.ObjectModel'
+local t = require 'telem.lib.util'
+
+local InputAdapter      = require 'telem.lib.InputAdapter'
+local Metric            = require 'telem.lib.Metric'
+local MetricCollection  = require 'telem.lib.MetricCollection'
+
+local BioGeneratorInputAdapter = o.class(InputAdapter)
+BioGeneratorInputAdapter.type = 'BioGeneratorInputAdapter'
+
+function BioGeneratorInputAdapter:constructor (peripheralName, categories)
+    self:super('constructor')
+
+    -- TODO this will be a configurable feature later
+    self.prefix = 'mekbiogen:'
+
+    -- TODO make these constants
+    local allCategories = {
+        'basic',
+    }
+
+    if not categories then
+        self.categories = { 'basic' }
+    elseif categories == '*' then
+        self.categories = allCategories
+    else
+        self.categories = categories
+    end
+
+    -- boot components
+    self:setBoot(function ()
+        self.components = {}
+
+        self:addComponentByPeripheralID(peripheralName)
+    end)()
+end
+
+function BioGeneratorInputAdapter:read ()
+    self:boot()
+
+    local source, generator = next(self.components)
+
+    local metrics = MetricCollection()
+
+    local loaded = {}
+
+    for _,v in ipairs(self.categories) do
+        -- skip, already loaded
+        if loaded[v] then
+            -- do nothing
+
+        -- Literally all we have lmao
+        elseif v == 'basic' then
```

Comment:
I think to preserve API parity, `fuel_*` should be changed to `bio_fuel_*`.

Categories should be expanded:

Basic:
- energy_filled_percentage
- bio_fuel_filled_percentage
- production_rate

Fuel:
- bio_fuel_capacity
- bio_fuel
- bio_fuel_needed

Energy:
- energy

Reply:
blehhh  my responses are gonna get sillier the more send it: 6814f0d031703d0c4c69997a9e64c5bd46dbbc9f